common/psbt_internal.c

```diff
@@ -35,18 +35,16 @@ void psbt_finalize_input(const tal_t *ctx,
 	 * on these just .. ignores it!? Murder. Anyway, here we do a final
 	 * scriptsig check -- if there's a redeemscript field still around we
 	 * just go ahead and mush it into the final_scriptsig field. */
-	if (in->redeem_script) {
+	const struct wally_map_item *redeem_script = wally_map_get_integer(&in->psbt_fields, /* PSBT_IN_REDEEM_SCRIPT */ 0x04);
```

Comment:
```suggestion
	const struct wally_map_item *redeem_script;
	 ..... 
	redeem_script = wally_map_get_integer(&in->psbt_fields, /* PSBT_IN_REDEEM_SCRIPT */ 0x04);
```

we can declare the struct before and then assign it? for a while I stay around the core lightning code base the previous code looks odd to me?

Reply:
fixed